src/Silex/ControllerCollection.php

```diff
@@ -57,11 +59,21 @@ public function __construct(Route $defaultRoute, $routesFactory = null)
     /**
      * Mounts controllers under the given route prefix.
      *
-     * @param string               $prefix      The route prefix
-     * @param ControllerCollection $controllers A ControllerCollection instance
+     * @param string                        $prefix      The route prefix
+     * @param ControllerCollection|callable $controllers A ControllerCollection instance or a callable for defining routes
+     *
+     * @throws \LogicException
      */
-    public function mount($prefix, ControllerCollection $controllers)
+    public function mount($prefix, $controllers)
     {
+        if (is_callable($controllers)) {
+            $collection = $this->controllersFactory ? call_user_func($this->controllersFactory) : new static(new Route(), new RouteCollection());
+            call_user_func($controllers, $collection);
+            $controllers = $collection;
```

Comment:
That's what I started to do, but then I realized that if we only did it in the application mount then the following code would not work:

```php
$app->mount('/admin', function($admin) {
    $admin->mount('/blog', function($blog) {
        $blog->get('/');
    });
});
```

instead, you'd have to do

```php
$app->mount('/admin', function($admin) use ($app) {
    // either we don't allow recursive routing, which might seem inconsistent for users
    // or we do something like this
    $app->mount($admin, '/blog', function($blog) {
        $blog->get('/');
    });
});
```